hashintel/hash
# Upgrade to next 13

## 🌟 What is the purpose of this PR?

Upgrades to NextJS 13 in `packages/hash/frontend` and `apps/hashdotdev`

The motivation for doing so now is that I want to add some middleware in the HASH frontend that can return a response directly, which is an experimental feature in 13. This to follow in a later PR.

## 🔍 What does this change?

- Upgrade deps
- Applies the result of running [codemods](https://beta.nextjs.org/docs/upgrade-guide/codemods#version-13) to convert existing `Image` and `Link` usage to its legacy pre-13 format. I tried running the codemod to automatically update `Image` to the 13 version (changes props etc) but it had weird results (e.g. left a prop on an `Image` that is removed in the new version) so we can do it as we touch that code. These changes should keep behavior the same.
- Remove the second webpack entrypoint for the sandbox code in the HASH frontend as it wasn't compiling, and isn't used at the moment anyway
- Clean up some unused code in `next.config.js` – we no longer use `@rjsf/material-ui` (which had some config applied to handle) or the SASS variables that were listed there

## ❓ How to test this?

1.  Check linting
2. Check no unexpected behavior / visual changes in preview deployments for hash.dev / HASH frontend
